src/video/x11/SDL_x11messagebox.c

```diff
@@ -305,10 +307,14 @@ static bool X11_ShowMessageBoxImpl(const SDL_MessageBoxData *messageboxdata, int
     }
 
     /* Create window */
+    if (messageboxdata->window && SDL_strcmp(viddev->name, "x11") == 0) {
```

Comment:
```suggestion
    if (messageboxdata->window && video && SDL_strcmp(video->name, "x11") == 0) {
```

Reply:
and done.

Reply:
Not done: video != NULL still not checked.